### ISSUE TITLE
Security Issue - Private/Public Keypair Exchange for External Clients

### ISSUE THREAD

**Issue by DrDoug88** [gravitl/netmaker#1378]
### What happened?

When creating external clients, the Netmaker server will automatically create the public/private keypair on behalf of the client. This is not how the public/private key exchange is supposed to work. The private key should be generated on the client side and only public keys exchanged. I realize that this is done in terms on convenience, but room should be made in order to perform a proper key exchange. Can the UI be updated so that the client's public key can be entered/edited? Also, I see a feature request to allow for Preshared Keys (https://github.com/gravitl/netmaker/issues/1231), can this also be done at the same time? Perhaps even allowing any of the external client configuration details to be edited?

### Version

v0.14.5

### What OS are you using?

Linux

### Contributing guidelines

- [X] Yes, I did.

**Comment by mattkasun** [comment]
> 

Yes, I answered too quickly .... didn't realize you were talking about ext clients

**Comment by DrDoug88** [comment]
No worries.

**Comment by 0xdcarns** [comment]
We will look to roadmap this as it would be a fairly straightforward security enhancement

**Comment by DrDoug88** [comment]
Okay great, thank your the prompt response. I'll keep an eye out for it.

**Comment by mattkasun** [comment]
PR #2100 and PR #2148 implement
